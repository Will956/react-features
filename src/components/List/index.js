import React from 'react';
import { string, arrayOf, object } from 'prop-types';

import ErrorBoundaries from '../../hoc/ErrorBoundaries';

const List = ({ title, content, list }) => (
  <>
    <h1>{title}</h1>
    <p>{content}</p>
    <ul>
      {list.map(item => <li key={item.content}>{item.content}</li>)}
    </ul>
  </>
);

List.propTypes = {
  title: string.isRequired,
  content: string.isRequired,
  list: arrayOf(object)
};

export default ErrorBoundaries(List);