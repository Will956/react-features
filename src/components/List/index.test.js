import React from 'react';
import { cleanup, render } from 'react-testing-library';
import List from './index';

global.console = {
  error: jest.fn()
};

const listProps = {
  title: 'my title',
  content: 'my content',
  list: [
    {
      content: 'a'
    },
    {
      content: 'b'
    }
  ]
};

afterEach(cleanup);

describe('List', () => {
  it('renders List with props', () => {
    const component = render(<List {...listProps}/>);
    expect(component).toMatchSnapshot();
  });
});