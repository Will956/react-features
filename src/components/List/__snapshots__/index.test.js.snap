// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders List with props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        my title
      </h1>
      <p>
        my content
      </p>
      <ul>
        <li>
          a
        </li>
        <li>
          b
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <h1>
      my title
    </h1>
    <p>
      my content
    </p>
    <ul>
      <li>
        a
      </li>
      <li>
        b
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
